storage/internal/elastic/config.go

```diff
@@ -0,0 +1,119 @@
+	// This is the maximum size (in kB) of bulk request sent while migrating data
+	maxBulkSize int `json:"max_bulk_size" default:"4000"`
+	// This is the maximum size (in term of number of documents) of bulk request sent while migrating data
+	maxBulkCount int `json:"max_bulk_count" default:"1000"`
+}
+
+// Get the tag for this field (for internal usage only: fatal if not found !).
+func getElasticStorageConfigPropertyTag(fn string, tn string) string {
+	t := reflect.TypeOf(elasticStoreConf{})
+	f, found := t.FieldByName(fn)
+	if !found {
+		log.Fatalf("Not able to get field %s on elasticStoreConf struct, there is an issue with this code !", fn)
+	}
+	tv := f.Tag.Get(tn)
+	if tv == "" {
+		log.Fatalf("Not able to get field %s's tag %s value on elasticStoreConf struct, there is an issue with this code !", fn, tn)
+	}
+	return tv
+}
+
+// The configuration of the elastic store is defined regarding default values and store 'properties' dynamic map.
+// Will fail if the required es_urls is not set in store 'properties'
+func getElasticStoreConfig(storeConfig config.Store) elasticStoreConf {
+	var cfg elasticStoreConf
+
+	storeProperties := storeConfig.Properties
+
+	// The ES urls is required
+	k := getElasticStorageConfigPropertyTag("esUrls", "json")
+	if storeProperties.IsSet(k) {
+		cfg.esUrls = storeProperties.GetStringSlice(k)
+		if cfg.esUrls == nil || len(cfg.esUrls) == 0 {
+			log.Fatalf("Not able to get ES configuration for elastic store, es_urls store property seems empty : %+v", storeProperties.Get(k))
+		}
+	} else {
+		log.Fatal("Not able to get ES configuration for elastic store, es_urls store property should be set !")
+	}
+
+	// Define store optional / default configuration
+	k = getElasticStorageConfigPropertyTag("caCertPath", "json")
+	if storeProperties.IsSet(k) {
+		cfg.caCertPath = storeProperties.GetString(k)
+	}
+	k = getElasticStorageConfigPropertyTag("esForceRefresh", "json")
+	if storeProperties.IsSet(k) {
+		cfg.esForceRefresh = storeProperties.GetBool(k)
+	} else {
+		cfg.esForceRefresh = cast.ToBool(getElasticStorageConfigPropertyTag("esForceRefresh", "default"))
+	}
+	k = getElasticStorageConfigPropertyTag("indicePrefix", "json")
+	if storeProperties.IsSet(k) {
+		cfg.indicePrefix = storeProperties.GetString(k)
+	} else {
+		cfg.indicePrefix = getElasticStorageConfigPropertyTag("indicePrefix", "default")
+	}
+	cfg.esQueryPeriod = getDurationFromSettingsOrDefaults("esQueryPeriod", storeProperties)
+	cfg.esRefreshWaitTimeout = getDurationFromSettingsOrDefaults("esRefreshWaitTimeout", storeProperties)
+	cfg.maxBulkSize = getIntFromSettingsOrDefaults("maxBulkSize", storeProperties)
+	cfg.maxBulkCount = getIntFromSettingsOrDefaults("maxBulkCount", storeProperties)
+
+	return cfg
+}
+
+// Get the duration from store config properties, fallback to required default value defined in struc.
+func getDurationFromSettingsOrDefaults(fn string, dm config.DynamicMap) time.Duration {
```

Comment:
this should be here: https://github.com/ystia/yorc/blob/develop/config/config.go as generic function

Reply:
I'm not sure to well understand. 
My need is to get default values that I decided to put in my `elasticStoreConf` fields tags 'default' for more readability. 
If the value is not found in DynamicMap, I read from it the 'default' tag. If I move this into config.go maybe I will need to make reflection on an interface{} and pass the type ?

Reply:
Yep, you're right. Not the same use of DynamicMap functions.